api/v1beta1/cluster_types.go

```diff
@@ -72,6 +72,24 @@ type ClusterSpec struct {
 	// this feature is highly experimental, and parts of it might still be not implemented.
 	// +optional
 	Topology *Topology `json:"topology,omitempty"`
+
+	// availabilityGates specifies additional conditions to include when evaluating Cluster availability.
+	// A Cluster is available if:
+	// * Cluster's `RemoteConnectionProbe` and `TopologyReconciled` conditions are true and
+	// * the control plane `Available` condition is true and
+	// * all worker resource's `Available` conditions are true and
+	// * all conditions defined in AvailabilityGates are true as well.
+	// +optional
+	// +listType=map
+	// +listMapKey=conditionType
+	AvailabilityGates []ClusterAvailabilityGate `json:"availabilityGates,omitempty"`
```

Comment:
Can we set a sensible maximum size for this list? 32 perhaps? (That's way more conditions than we currently have so should be future proof?)

Reply:
Add, btw we have a note for v1beta2 API to check this stuff all around.